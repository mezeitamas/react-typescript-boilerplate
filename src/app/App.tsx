import React from 'react';
import type { FunctionComponent, ReactElement } from 'react';

import { Dummy } from '@/shared';

const App: FunctionComponent = (): ReactElement => {
    return (
        <main role="main">
            <h1>Hello, world!</h1>

            <Dummy
                name="Dummy"
                upperCase={true}
            />
        </main>
    );
};

export { App };
