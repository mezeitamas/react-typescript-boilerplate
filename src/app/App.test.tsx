import React from 'react';

import { describe, test, expect } from '@rstest/core';
import { render } from '@testing-library/react';

import { App } from './App';

describe('<App />', () => {
    test('should render', () => {
        // Given

        // When
        const { asFragment } = render(<App />);

        // Then
        expect(asFragment()).toMatchSnapshot();
    });
});
