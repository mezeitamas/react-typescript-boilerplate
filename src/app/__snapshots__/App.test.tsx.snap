// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render 1`] = `
<main
  role="main"
>
  <h1>
    Hello, world!
  </h1>
  <Dummy
    name="Dummy"
    upperCase={true}
  />
</main>
`;
